godel_process_planning/src/blend_process_planning.cpp

```diff
@@ -46,91 +40,101 @@ static inline descartes_core::TrajectoryPtPtr toDescartesPt(const Eigen::Affine3
                                               AxialSymmetricPt::Z_AXIS, tm);
 }
 
-/**
- * @brief Computes a 'cost' value for a robot motion between 'source' and 'target'
- * @param source  The joint configuration at start of motion
- * @param target  The joint configuration at end of motion
- * @return cost value
- */
-double freeSpaceCostFunctionBlending(const std::vector<double>& source,
-                                     const std::vector<double>& target)
+static EigenSTL::vector_Affine3d linearMoveZ(const Eigen::Affine3d& origin, double step_size, int steps)
 {
-  // The cost function here penalizes large single joint motions in an effort to
-  // keep the robot from flipping a joint, even if some other joints have to move
-  // a bit more.
-  double cost = 0.0;
-  for (std::size_t i = 0; i < source.size(); ++i)
+  EigenSTL::vector_Affine3d result (steps);
+
+  for (int i = 0; i < steps; ++i)
   {
-    double diff = std::abs(source[i] - target[i]);
-    if (diff > FREE_SPACE_MAX_ANGLE_DELTA)
-      cost += FREE_SPACE_ANGLE_PENALTY * diff;
-    else
-      cost += diff;
+    result[i] = origin * Eigen::Translation3d(0, 0, step_size * (i + 1));
   }
-  return cost;
+
+  return result;
 }
 
+
+
 /**
- * @brief transforms an input, in the form of a reference pose and points relative to that pose,
- * into Descartes'
- *        native format. Also adds in associated parameters.
- * @param ref The reference posed that all points are multiplied by. Should be in the world space of
- * the blend move group.
- * @param points Sequence of points (relative to ref and the world space of blending robot model)
- * @param params Surface blending parameters, including info such as traversal speed
- * @return The input trajectory encoded in Descartes points
+ * @brief toDescartesTraj
+ * @param poses
+ * @param params
+ * @return
  */
 static godel_process_planning::DescartesTraj
-toDescartesTraj(const geometry_msgs::Pose& ref, const std::vector<geometry_msgs::Point>& points,
+toDescartesTraj(const std::vector<geometry_msgs::PoseArray>& segments,
                 const godel_msgs::BlendingPlanParameters& params)
 {
-  DescartesTraj traj;
-  traj.reserve(points.size());
-  if (points.empty())
-    return traj;
+  std::vector<EigenSTL::vector_Affine3d> connections;
 
-  Eigen::Affine3d last_pose = createNominalTransform(ref, points.front());
+  const static double step_size = 0.02; // m
```

Comment:
I strongly dislike that all caps convention: macros should be all caps not normal variables. As far as the locality, I argue that the constant should go in the smallest scope possible. Since this is only used in the function, I'd keep it there for parse-ability.